common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/LocalStatisticsDownloadService.java

```diff
@@ -0,0 +1,72 @@
+package app.coronawarn.server.common.persistence.service;
+
+import static java.time.ZoneOffset.UTC;
+
+import app.coronawarn.server.common.persistence.domain.LocalStatisticsDownload;
+import app.coronawarn.server.common.persistence.repository.LocalStatisticsDownloadRepository;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+public class LocalStatisticsDownloadService {
+
+  private final LocalStatisticsDownloadRepository repository;
+  private static final Logger logger = LoggerFactory.getLogger(LocalStatisticsDownloadService.class);
+
+  public LocalStatisticsDownloadService(LocalStatisticsDownloadRepository repository) {
+    this.repository = repository;
+  }
+
+  /**
+   * Stores an entry in the Database with {@param timestamp} as seconds and {@param eTag}. The counter is an incremental
+   * value automatically determined by a postgres sequence.
+   *
+   * @param timestamp in seconds.
+   * @param etag      value to be stored.
+   * @return boolean determining if store was successful or not.s
+   */
+  @Transactional
+  public boolean store(long timestamp, String etag) {
+    try {
+      this.repository.insertWithAutoIncrement(timestamp, etag);
+      return true;
+    } catch (Exception e) {
+      logger.error("Failed to store Statistics Download entry", e);
+      return false;
+    }
+  }
+
+  /**
+   * Retrieves the latest {@link LocalStatisticsDownload} stored. The order is determined by the counter field, which is
+   * automatically incremented by the Database.
+   *
+   * @return {@link LocalStatisticsDownload} returns Optional.empty if no download entries are stored.
+   */
+  public Optional<LocalStatisticsDownload> getMostRecentDownload() {
+    return Optional.ofNullable(repository.getWithLatestETag());
```

Comment:
What do you think of moving the Optional into the Repository? So basicall y in the repository return   Optional<LocalStatisticsDownload>

Reply:
I agree. I moved it.